### PR TITLE
Refactor a bit

### PR DESCRIPTION
**Pull request by okurz**
* Make test_approve not rely on specific log level selection
* Refactor too big for-loop in types/incidents.py

### CODE REVIEW

File: openqabot/types/incidents.py [hunk 1]
```diff
@@ -76,204 +76,215 @@ def _is_scheduled_job(
...
-                    # parse Public-Cloud pint query if present
-                    if "PUBLIC_CLOUD_PINT_QUERY" in settings:
-                        settings = apply_publiccloud_pint_image(settings)
-                        if not settings.get("PUBLIC_CLOUD_IMAGE_ID", False):
-                            continue
-
-                    full_post["openqa"] = settings
-                    full_post["qem"]["settings"] = settings
-                    ret.append(full_post)
+        ret = list(filter(None, ret))
```
Comment (mimi1vx) [inline openqabot/types/incidents.py:289]: please no ... this is awful and really discouraged python syntax

`ret = [r for r in ret if r]`

**Review by mimi1vx — CHANGES_REQUESTED** [review]

**Comment by kalikiana** [comment]
I would prefer **Make test_approve not rely on specific log level selection** in its own PR rather than unrelated "bits"

**Comment by mimi1vx** [comment]
> I would prefer **Make test_approve not rely on specific log level selection** in its own PR rather than unrelated "bits"

ah, its in different commits ...

**Review by Martchus — APPROVED** [review]

**Comment by okurz** [comment]
> I would prefer **Make test_approve not rely on specific log level selection** in its own PR rather than unrelated "bits"

-> #150

File: openqabot/types/incidents.py (hunk 1 above)
Comment (okurz) [reply openqabot/types/incidents.py]: thank you. That's actually also what I prefer. Done

**Comment by okurz** [comment]
> https://github.com/openSUSE/qem-bot/actions/runs/6586389281/job/17894634676?pr=149#step:6:17
> 
> ```
> ************* Module openqabot.types.incidents
> openqabot/types/incidents.py:79:4: R0911: Too many return statements (13/6) (too-many-return-statements)
> ```

Yeah, I guess that was expected due to the still many abort paths in the original code. I wonder why that's not an error aborting the CI run however I suggest we accept this refactoring first

**Comment by a-kpappas** [comment]
This is the perfect opportunity to add some comments. What does 'handle' mean?

**Comment by okurz** [comment]
> This is the perfect opportunity to add some comments. What does 'handle' mean?

It means "do stuff" as I don't have a better clue what it actually does ;)

**Comment by okurz** [comment]
@mimi1vx do you request further changes or can approve?

**Review by mimi1vx — APPROVED** [review]

**Pull request merged**